src/code/relocation.c

```diff
@@ -37,55 +37,60 @@ void Overlay_Relocate(void* allocatedVRamAddress, OverlayRelocationSection* over
 
     for (i = 0; i < overlayInfo->nRelocations; i++) {
         reloc = overlayInfo->relocations[i];
-        relocDataP = (u32*)(sections[reloc >> 0x1E] + (reloc & 0xFFFFFF));
+        relocDataP = (u32*)(sections[RELOC_SECTION(reloc)] + RELOC_OFFSET(reloc));
         relocData = *relocDataP;
-        switch (reloc & 0x3F000000) {
-            case 0x2000000:
-                /* R_MIPS_32
-                 * Handles 32-bit address relocation.  Used in things such as
-                 * jump tables.
-                 */
+
+        switch (RELOC_TYPE(reloc)) {
+            case R_MIPS_32 << 24:
+                // Handles 32-bit address relocation, used for things such as jump tables and pointers in data.
+                // Just relocate the full address.
+
+                // Check address is valid for relocation
                 if ((*relocDataP & 0xF000000) == 0) {
```

Comment:
not your change but
```suggestion
                if ((*relocDataP & 0x0F000000) == 0) {
```
would be easier to read imo / less confusion-prone
(same with `0x3FFFFFF` and `0xFFFFFFF` later I guess)

I'm not sure what this check is meant for to be honest, could have something to do with segmented addresses but those symbols aren't in relocation data obviously

Reply:
It is a sanity check to verify that the address is actual valid for relocation (see comment above the if statement). See https://discord.com/channels/688807550715560050/688850564595187788/965708627165970462, Tharo's message, which explains why this is the case.

Reply:
idk I don't see why the vram address couldn't go past 0x80FFFFFF in theory, overlays currently start at vram `0x80800000` but why not `0x81000000`, if it wasn't for this piece of code